src/foam/dao/index/TreeNode.java

```diff
@@ -345,4 +357,24 @@ public TreeNode lte(TreeNode s, Object key, PropertyInfo prop) {
       s.level, s.left, null);
   }
 
+  public Object[] getAllValues(Object s){
```

Comment:
Getting all values uses up memory for the Object[]. Passing in a sink lets each row be processed in sequence without accumulating any extra memory. So a better method would be:
  public void select(Object state, Sink sink)